functions-v2/src/at-midnight.ts

```diff
@@ -0,0 +1,28 @@
+import {onSchedule} from "firebase-functions/v2/scheduler";
+import * as logger from "firebase-functions/logger";
+
+// NOTE: in order for this import from shared to work it is necessary
+// to alias "firebase-admin" in tsconfig.json. Otherwise Typescript will
+// read the types from the parent node_modules. The parent directory
+// has a different version of the firebase dependencies, which cause
+// type errors.
+import {cleanFirebaseRoots} from "../../shared/clean-firebase-roots";
+
+export const atMidnight = onSchedule("0 0 * * *", runAtMidnight);
```

Comment:
Midnight GMT is just 4pm PST.  I know it waits 24 hours before actually deleting things, but still might make more sense to push this to, say, 7am GMT so it's more the middle of the night across the US.